mlflow/metrics/utils/metric_definitions.py

```diff
@@ -0,0 +1,409 @@
+    except Exception as e:
+        _logger.warning(
+            f"Failed to load 'toxicity' metric (error: {e!r}), skipping metric logging."
+        )
+        return
+
+    _logger.info("Computing toxicity metric:")
+    scores = toxicity.compute(predictions=predictions)["toxicity"]
+    toxicity_ratio = toxicity.compute(predictions=predictions, aggregation="ratio")[
+        "toxicity_ratio"
+    ]
+    return MetricValue(scores=scores, aggregate_results={"ratio": toxicity_ratio})
+
+
+def _perplexity_eval_fn(eval_df, metrics):
+    y_pred = eval_df["prediction"]
+    predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+
+    if not _validate_text_predictions(predictions, "perplexity"):
+        return
+
+    try:
+        _logger.info("Loading perplexity metric:")
+        import evaluate
+
+        perplexity = evaluate.load("perplexity", module_type="metric")
+    except Exception as e:
+        _logger.warning(
+            f"Failed to load 'perplexity' metric (error: {e!r}), skipping metric logging."
+        )
+        return
+
+    _logger.info("Computing perplexity metric:")
+    results = perplexity.compute(predictions=predictions, model_id="gpt2")
+    return MetricValue(
+        scores=results["perplexities"], aggregate_results={"mean": results["mean_perplexity"]}
+    )
+
+
+def _flesch_kincaid_eval_fn(eval_df, metrics):
+    y_pred = eval_df["prediction"]
+    predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+
+    if not _validate_text_predictions(predictions, "flesch_kincaid"):
+        return
+
+    try:
+        import textstat
+    except ImportError:
+        _logger.warning("Failed to load flesch kincaid metric, skipping metric logging.")
+        return
+
+    _logger.info("Computing flesch kincaid metric:")
+    scores = [textstat.flesch_kincaid_grade(prediction) for prediction in predictions]
+    return MetricValue(scores=scores, aggregate_results={"mean": sum(scores) / len(scores)})
+
+
+def _ari_eval_fn(eval_df, metrics):
+    y_pred = eval_df["prediction"]
+    predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+
+    if not _validate_text_predictions(predictions, "ari"):
+        return
+
+    try:
+        import textstat
+    except ImportError:
+        _logger.warning(
+            "Failed to load automated readability index metric, skipping metric logging."
+        )
+        return
+
+    _logger.info("Computing automated readability index metric:")
+    scores = [textstat.automated_readability_index(prediction) for prediction in predictions]
+    return MetricValue(scores=scores, aggregate_results={"mean": sum(scores) / len(scores)})
+
+
+def _accuracy_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        from sklearn.metrics import accuracy_score
+
+        acc = accuracy_score(y_true=eval_df["target"], y_pred=eval_df["prediction"])
+        return MetricValue(aggregate_results={"": acc})
```

Comment:
is this supposed to be an empty string?

Reply:
Yes, wasn't sure what to else to put here considering there are no scores and it is just an aggregation - I was thinking that metrics will be named metric_name_aggregation_name/version so an empty string would mean that it just takes the metric name.